src/server/components/lists/api/controllers.ts

```diff
@@ -0,0 +1,45 @@
+import type { Request, Response } from "express";
+import Joi from "joi";
+import { prisma } from "server/models/db/prisma-client";
+import { logger } from "server/services/logger";
+
+export async function listsApiPostController(req: Request, res: Response) {
+  const receivedData = {
+    type: req.body.type as "funeralDirectors" | "translatorsInterpreters" | "lawyers",
+    country: req.body.country as string,
+  };
+
+  const { error } = receivedDataSchema.validate(receivedData);
+
+  if (error) {
+    logger.error(`listsApiPostController schema validation error: ${error.details[0].message}`);
+    res.status(400);
+    return;
+  }
+
+  try {
+    const result = await prisma.list.findFirst({
+      where: {
+        type: receivedData.type,
+        country: {
+          name: receivedData.country,
+        },
+      },
+    });
+
+    if (!result) {
+      res.sendStatus(404);
```

Comment:
hmm, I think 400 (bad request) might actually be slightly more appropriate here rather than 404 (but it's not that big of a deal). I've seen 404 used in this way

404 usually refers to the endpoint (/api/lists) rather than the type with country. 

404 suggests /api/lists doesn't exist, 400 suggests the request was received but erroneous in some way. 

Might also be worth sending across a message as well, so we can log it on the runner.

```
{ status: 400, message: `List for ${type} ${country} not found` }
```